common/src/test/java/io/netty/util/concurrent/GlobalEventExecutorTest.java

```diff
@@ -91,6 +94,29 @@ public void testScheduledTasks() throws Exception {
         assertThat(e.thread, sameInstance(thread));
     }
 
+    // ensure that when a task submission causes a new thread to be created, the thread inherits the thread group of the
+    // submitting thread
+    @Test(timeout = 2000)
+    public void testThreadGroup() throws InterruptedException {
+        final ThreadGroup group = new ThreadGroup("group");
+        final AtomicReference<ThreadGroup> capturedGroup = new AtomicReference<ThreadGroup>();
+        final Thread thread = new Thread(group, new Runnable() {
+            @Override
+            public void run() {
+                final Thread t = e.threadFactory.newThread(new Runnable() {
+                    @Override
+                    public void run() {
+                    }
+                });
+                capturedGroup.set(t.getThreadGroup());
+            }
+        });
+        thread.start();
+        thread.join();
+
+        assertThat(capturedGroup.get(), equalTo(group));
```

Comment:
I think we raced, I pushed 8fdd05202fea5a6b3a59100ebe47589ca059a3fc already. :smile: